### PR TITLE
TAM1

### PR DESCRIPTION
**Pull request by valeria1604**

### CODE REVIEW

File: src/com/company/TAM1/Main.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.Arrays;
+
+public class Main {
+
+    public static final String SPECIFIED_CHARACTER = "w";
+
+    public static void main(String[] args) {
+        String sentence1 = "Width world Wide web";
+        String sentence2 = "WWW? Four-bedroom farmhouse in the countryside. Wave! All of the four double bedrooms are en suite.";
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]: This task is about working with file. Please add the reading data from the file. File examples present in the task description

File: src/com/company/TAM1/Main.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package com.company.TAM1;
+
+import java.util.Arrays;
+
+public class Main {
+
+    public static final String SPECIFIED_CHARACTER = "w";
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]:
```suggestion
    public static final String FILTER_BY_CHARACTER = "w";
```

File: src/com/company/TAM1/Main.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        String sentence1 = "Width world Wide web";
+        String sentence2 = "WWW? Four-bedroom farmhouse in the countryside. Wave! All of the four double bedrooms are en suite.";
+
+        System.out.println(Arrays.asList(getSentenceArray(sentence1)));
+        System.out.println(Arrays.asList(getSentenceArray(sentence2)));
+    }
+
+    private static String[] getSentenceArray(String sentence) {
+        sentence = sentence.toLowerCase();
+        String[] splintedSentence = sentence.split("\\W+");
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]:
```suggestion
        String[] splitedSentence = sentence.split("\\W+");
```

File: src/com/company/TAM1/Main.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                wordCount++;
+            }
+        }
+
+        String[] finalResult = new String[wordCount];
+        int index = 0;
+
+        for (String word : splintedSentence) {
+            if (word.startsWith(SPECIFIED_CHARACTER)) {
+                finalResult[index] = word;
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]: please also implement result sorting that was mentioned in the task description

**Review by maxline — CHANGES_REQUESTED** [review]
good job! let's improve the solution

File: src/com/company/TAM1/Main.java (hunk 2 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 3 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 1 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 4 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package com.company.TAM1;
+
+import java.util.Arrays;
+
+public class Main {
+
+    public static final String FILTER_BY_CHARACTER = "w";
+
+    public static void main(String[] args) {
+        String fromFileName =  "test.txt";
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]: formatting, remove extra spaces please.
also add the file `test.txt` to the PR

File: src/com/company/TAM1/Main.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        StringBuilder text = new StringBuilder();
+        String line;
+
+        try(
+                BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+        ) {
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]: please fix formatting

File: src/com/company/TAM1/Main.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            while ((line = reader.readLine()) != null){
+                text.append(line);
+            }
+
+        }catch (IOException e) {
+            e.printStackTrace();
+        }
+        String[] result = getSentenceArray(String.valueOf(text));
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]:
```suggestion
        String[] result = getSentenceArray(builder.toString());
```

File: src/com/company/TAM1/Main.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.Arrays;
+
+public class Main {
+
+    public static final String FILTER_BY_CHARACTER = "w";
+
+    public static void main(String[] args) {
+        String fromFileName =  "test.txt";
+        StringBuilder text = new StringBuilder();
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]:
```suggestion
        StringBuilder builder = new StringBuilder();
```

**Review by maxline — CHANGES_REQUESTED** [review]
good job! a couple of more suggestions

File: src/com/company/TAM1/Main.java (hunk 5 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 6 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 8 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java (hunk 7 above)
Comment (valeria1604) [reply src/com/company/TAM1/Main.java]: fixed

File: src/com/company/TAM1/Main.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        try (
+                BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+        ) {
+            while ((line = reader.readLine()) != null) {
+                builder.append(line);
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
Comment (maxline) [inline src/com/company/TAM1/Main.java]: 1 let's move the code that retrieving string from file in the separate method String readFromFile(String fileName)
2 do the path to fileName chosen correctly? I see that you added file `src/main/resources/test.txt`

File: test.txt [hunk 10]
```diff
@@ -0,0 +1,6 @@
+Beautiful two-bedroom city flat five minutes' walk from the cathedral.
```
Comment (maxline) [inline test.txt]: Why do we need duplicate the file?
We already have: `src/main/resources/test.txt`

**Review by maxline — CHANGES_REQUESTED** [review]
Good job! Added a couple of more comment

**Review by maxline — APPROVED** [review]
Well done

**Pull request merged**